src/identity/identity-wallet.ts

```diff
@@ -858,16 +909,53 @@ export class IdentityWallet implements IIdentityWallet {
     );
   }
 
-  private getKMSIdByAuthCredential(credential: W3CCredential): KmsKeyId {
-    if (credential.type.indexOf('AuthBJJCredential') === -1) {
-      throw new Error("can't sign with not AuthBJJCredential credential");
+  /** {@inheritDoc IIdentityWallet.publishStateToReverseHashService} */
+  async publishStateToReverseHashService(opts: RHSOptions): Promise<void> {
+    if (!opts.issuerDID && !opts.treeModel) {
```

Comment:
issuerDID should be mandatory.
decisison is based on treeModel
publishStateToReverseHashService  is  publishToCredentialRegistry